### PR TITLE
Deprecate clamp!

### LINKED ISSUE #22236: Deprecate `clamp!(A, lo, hi)` for `broadcast!(clamp, A, A, lo, hi)`?
Should we deprecate `clamp!(A, lo, hi)` for `broadcast!(clamp, A, A, lo, hi)`? Note the latter can be written `A .= clamp.(A, lo, hi)`.

Related: https://github.com/JuliaLang/julia/pull/22235#discussion_r120257450

Part of #20402

### PR DESCRIPTION
**Pull request by arghhhh**
Fixes #22236

### CODE REVIEW

**Comment by fredrikekre** [comment]
Should be removed here too:
https://github.com/JuliaLang/julia/blob/80369bf7b49959d28837e764b41621456ae0e32e/base/exports.jl#L445
https://github.com/JuliaLang/julia/blob/80369bf7b49959d28837e764b41621456ae0e32e/base/math.jl#L15

**Comment by arghhhh** [comment]
There's a reason for my github name!  OK.  I have removed these two mentions of `clamp!`, and grep'ed the rest of source code.  There is an unrelated `clamp!` in base/grisu/bignums.

File: base/deprecated.jl [hunk 1]
```diff
@@ -748,6 +748,7 @@ end
 
 # Deprecate manually vectorized clamp methods in favor of compact broadcast syntax
 @deprecate clamp(A::AbstractArray, lo, hi) clamp.(A, lo, hi)
+@deprecate clamp!(A::AbstractArray, lo, hi) A .= clamp.(A, lo, hi)
```
Comment (tkelman) [inline base/deprecated.jl]: this needs to be in the 0.7 section down near the end of the file

Comment (arghhhh) [inline base/deprecated.jl]: OK. I appreciate your patience with me.  Thanks.

**Comment by tkelman** [comment]
doc build is having trouble on CI, looks like you need to remove `doc/src/stdlib/math.md` line 134

**Comment by JeffBezanson** [comment]
This function is actually not a huge problem --- since `clamp!` only has a method for arrays, it can't be confused with a scalar function generalized to vectors.

**Comment by StefanKarpinski** [comment]
I agree. I find `clamp!(A, lo, hi)` quite convenient and replacing it with `A .= clamp(A, lo, hi)` while doable seems kind of unnecessary.

**Pull request closed without merging**